drivers/ipvlan/ipvlan_network.go

```diff
@@ -0,0 +1,163 @@
+func (d *driver) CreateNetwork(id string, option map[string]interface{}, ipV4Data, ipV6Data []driverapi.IPAMData) error {
+	// parse and validate the config and bind to networkConfiguration
+	config, err := parseNetworkOptions(id, option)
+	if err != nil {
+		return err
+	}
+	if id == "" {
+		return fmt.Errorf("Network ID cannot be empty")
+	}
+	config.ID = id
+	err = config.processIPAM(id, ipV4Data, ipV6Data)
+	if err != nil {
+		return err
+	}
+	// verify the macvlan mode from -o ipvlan_mode option
+	switch config.IpvlanMode {
+	case modeL2:
+		config.IpvlanMode = modeL2
+	case modeL3:
+		config.IpvlanMode = modeL3
+	case "":
+		// default to ipvlan L2 mode if -o ipvlan_mode is empty
+		config.IpvlanMode = modeL2
+	default:
+		return fmt.Errorf("requested ipvlan mode '%s' is not valid, 'l2' mode is the ipvlan driver default", config.IpvlanMode)
+	}
+	// user must specify a parent interface on the host to attach the container vif --host_iface=eth0
+	if config.HostIface == "" {
+		return fmt.Errorf("%s requires an interface from the docker host to be specified (usage: -o host_iface=eth)", ipvlanType)
+	}
+	d.createNetwork(config)
+	// update persistent db with the new network
+	return d.storeUpdate(config)
+}
+
+// createNetwork is used by new network callbacks and persistent network cache
+func (d *driver) createNetwork(config *configuration) error {
+	// if the host iface doesn't exist the requested pool will be released
+	if ok := validateHostIface(config.HostIface); !ok {
+		// if subinterface vlan checks do not pass, fail the network create
+		err := applyVlanAdd(config.HostIface)
+		if err != nil {
+			return err
+		}
+	}
+	n := &network{
+		id:        config.ID,
+		driver:    d,
+		endpoints: endpointTable{},
+		config:    config,
+	}
+	// update *network
+	d.addNetwork(n)
+	return nil
+}
+
+// DeleteNetwork the network for the specified driver type
+func (d *driver) DeleteNetwork(nid string) error {
+	if nid == "" {
+		return fmt.Errorf("network ID cannot be empty")
+	}
+	n := d.network(nid)
+	if n == nil {
+		return fmt.Errorf("could not find network with ID %s", nid)
+	}
+	// if the interface exists, ensure it is a slave device being deleted
+	if ok := validateHostIface(n.config.HostIface); ok {
+		err := verifyVlanDel(n.config.HostIface)
+		if err != nil {
+			logrus.Debugf("link %s was not deleted, continuing the delete network operation: %v", n.config.HostIface, err)
+		}
+	}
+	d.deleteNetwork(nid)
+	// delete the network record from persistent cache
+	return d.storeDelete(n.config)
+}
+
+// parseNetworkOptions parse docker network options
+func parseNetworkOptions(id string, option options.Generic) (*configuration, error) {
+	var (
+		err    error
+		config = &configuration{}
+	)
+	// Parse generic labels first
+	if genData, ok := option[netlabel.GenericData]; ok && genData != nil {
+		if config, err = parseNetworkGenericOptions(genData); err != nil {
+			return nil, err
+		}
+	}
+	return config, nil
```

Comment:
Can the `netlabel.Internal` option be honored with this driver ?
If not, we should fail the network creation if that option was passed.